ui/web.go

```diff
@@ -72,8 +73,16 @@ func Register(r *route.Router, reloadCh chan<- struct{}, logger log.Logger) {
 	))
 
 	r.Post("/-/reload", func(w http.ResponseWriter, req *http.Request) {
-		w.Write([]byte("Reloading configuration file..."))
-		reloadCh <- struct{}{}
+		errc := make(chan error)
+		defer close(errc)
+
+		reloadCh <- errc
```

Comment:
I don't think passing around channels like this is appropriate here, it's only ever going to be a single error received from `errc`.

Reply:
Coincidentally, it seems prometheus is using this pattern: https://github.com/prometheus/prometheus/blob/master/web/web.go#L638-L644

Reply:
I've been trying to think of a solution, but can't off the top of my head, so I think it's fine to go with it then.